aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2349,7 +2358,20 @@ impl AptosVM {
         let gas_used = Self::gas_used(max_gas_amount.into(), &gas_meter);
         match execution_result {
             Ok(result) => ViewFunctionOutput::new(Ok(result), gas_used),
-            Err(e) => ViewFunctionOutput::new(Err(e), gas_used),
+            Err(e) => {
+                let txn_status = TransactionStatus::from_vm_status(
+                    e.clone(),
+                    vm.features()
+                        .is_enabled(FeatureFlag::CHARGE_INVARIANT_VIOLATION),
+                );
+                let execution_status = match txn_status {
+                    TransactionStatus::Keep(status) => status,
+                    _ => ExecutionStatus::MiscellaneousError(Some(e.status_code())),
+                };
+                let status_with_abort_info =
```

Comment:
While this works for Move aborts (basically getting the error message defined as a comment), our beautiful error systems discards and messages when converted to VM status. I think we can keep partial VM error instead of status to return a message, and inject abort info conditionally only for Move aborts?

Reply:
refactored

Reply:
now this change should only affect aborts